test/synth/synth-stdout.test.ts

```diff
@@ -317,21 +323,714 @@ describe('validations', () => {
+      },
+    ],
+    [
+      withOnlyConfigInputs,
+      {
+        config: {
+          synthConfig: {
+            format: SynthesisFormat.HELM,
+            chartVersion: '1.1.1',
+            chartApiVersion: HelmChartApiVersion.V1,
+          },
+          imports: ['k8s'],
+        },
+        postSynth: synthWorksForChartAPIv1,
+      },
+    ],
+    [
+      withSameInputsInBoth,
+      {
+        format: SynthesisFormat.HELM,
+        chartVersion: '1.1.1',
+        chartApiVersion: HelmChartApiVersion.V1,
+        config: {
+          synthConfig: {
+            format: SynthesisFormat.HELM,
+            chartVersion: '1.1.1',
+            chartApiVersion: HelmChartApiVersion.V1,
+          },
+          imports: ['k8s'],
+        },
+        postSynth: synthWorksForChartAPIv1,
+      },
+    ],
+    [
+      withDifferentInputsInBoth,
+      {
+        format: SynthesisFormat.HELM,
+        chartVersion: '1.1.1',
+        chartApiVersion: HelmChartApiVersion.V1,
+        config: {
+          synthConfig: {
+            chartApiVersion: HelmChartApiVersion.V2,
+          },
+          imports: ['k8s'],
+        },
+        postSynth: synthWorksForChartAPIv1,
+      },
+    ],
+  ])('--chart-api-version is v1 %s', async (_testName, synthOptions) => {
+    await synth(synthOptions);
+  });
+
+  const synthWorksForChartAPIv2 = async (dir: string) => {
+    await expectSynthMatchSnapshot(dir.concat('/dist'));
+  };
+
+  test.each([
+    [
+      withOnlyCliInputs,
+      {
+        format: SynthesisFormat.HELM,
+        chartVersion: '1.1.1',
+        postSynth: synthWorksForChartAPIv2,
+      },
+    ],
+    [
+      withOnlyConfigInputs,
+      {
+        config: {
+          synthConfig: {
+            format: SynthesisFormat.HELM,
+            chartVersion: '1.1.1',
+          },
+        },
+        postSynth: synthWorksForChartAPIv2,
+      },
+    ],
+    [
+      withSameInputsInBoth,
+      {
+        format: SynthesisFormat.HELM,
+        chartVersion: '1.1.1',
+        config: {
+          synthConfig: {
+            format: SynthesisFormat.HELM,
+            chartVersion: '1.1.1',
+          },
+        },
+        postSynth: synthWorksForChartAPIv2,
+      },
+    ],
+    [
+      withDifferentInputsInBoth,
+      {
+        format: SynthesisFormat.HELM,
+        chartVersion: '1.1.1',
+        config: {
+          synthConfig: {
+            format: SynthesisFormat.PLAIN,
+            chartVersion: '1.1.1',
+          },
+        },
+        postSynth: synthWorksForChartAPIv2,
+      },
+    ],
+  ])('--chart-api-version is v2 without crds %s', async (_testName, synthOptions) => {
+    await synth(synthOptions);
+  });
+
+  const importsForChartApiv2 = [
+    'k8s',
+    path.join(__dirname, './__resources__/crds/foo.yaml'),
+    `bar:=${path.join(__dirname, './__resources__/crds/bar.yaml')}`,
+    'github:crossplane/crossplane@0.14.0',
+  ];
+
+  const synthWorksForChartAPIv2WithCrds = async (dir: string) => {
```

Comment:
Why do we need all these functions? The flow should just be: "synthesize -> expectSnapshot", without any specific comparison functions. No?